protocol/communities_messenger_signers_test.go

```diff
@@ -0,0 +1,246 @@
+	s.advertiseCommunityTo(s.john, community, s.alice)
+
+	s.joinCommunity(s.john, community, s.bob)
+	s.joinCommunity(s.john, community, s.alice)
+
+	// john as control node publishes community update
+	johnDescr := "john's description"
+	response, err := s.john.EditCommunity(&requests.EditCommunity{
+		CommunityID: community.ID(),
+		CreateCommunity: requests.CreateCommunity{
+			Name:        community.Name(),
+			Description: johnDescr,
+			Color:       community.Color(),
+			Membership:  protobuf.CommunityPermissions_NO_MEMBERSHIP,
+		},
+	})
+	s.Require().NoError(err)
+	s.Require().Equal(johnDescr, response.Communities()[0].Description().Identity.Description)
+
+	// bob accepts community update
+	_, err = WaitOnMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && r.Communities()[0].Description().Identity.Description == johnDescr
+		},
+		"no communities",
+	)
+	s.Require().NoError(err)
+
+	// alice accepts community update
+	_, err = WaitOnMessengerResponse(
+		s.alice,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && r.Communities()[0].Description().Identity.Description == johnDescr
+		},
+		"no communities",
+	)
+	s.Require().NoError(err)
+
+	// Alice will be transferred the ownership token, and alice will let others know
+
+	var chainID uint64 = 1
+	communityAddress := "community-address"
+	tokenAddress := "token-address"
+	tokenName := "tokenName"
+	tokenSymbol := "TSM"
+
+	// Update mock
+	// The signer for the community returned by the contracts should be alice
+	s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.alice.identity.PublicKey))
+	s.collectiblesServiceMock.SetMockCollectibleContractData(chainID, tokenAddress,
+		&communitytokens.CollectibleContractData{TotalSupply: &bigint.BigInt{}})
+
+	community, err = s.alice.communitiesManager.UpdatePrivateKeyAndControlNode(community.ID(), s.alice.identity)
+	s.Require().NoError(err)
+	s.Require().True(community.IsControlNode())
+
+	// Create community token
+	_, err = s.alice.SaveCommunityToken(&token.CommunityToken{
+		TokenType:       protobuf.CommunityTokenType_ERC721,
+		CommunityID:     community.IDString(),
+		Address:         tokenAddress,
+		ChainID:         int(chainID),
+		Name:            tokenName,
+		Supply:          &bigint.BigInt{},
+		Symbol:          tokenSymbol,
+		PrivilegesLevel: token.OwnerLevel,
+	}, nil)
+	s.Require().NoError(err)
+
+	err = s.alice.AddCommunityToken(community.IDString(), int(chainID), tokenAddress)
+	s.Require().NoError(err)
+
+	// make alice the control node
+	transaction := transactions.SendTxArgs{}
+	_, err = s.alice.SetCommunitySignerPubKey(context.Background(), community.ID(), chainID, communityAddress, transaction, "password", common.PubkeyToHex(&s.alice.identity.PublicKey))
+	s.Require().NoError(err)
+
+	var published bool
+
+	// We check the updated description is published as a signal
```

Comment:
Why do we check it as signal? Shouldn't John and Bob populate `MessengerResponse` with updated community in `RetrieveAll`?

Reply:
no, maybe I misunderstand the question, but in this case, the check is asynchronous (it's queued), and so it won't be published through RetrieveAll, but as a signal.
Did I understand the question correctly?

Reply:
Ah, you are right :+1:

FYI: I have created a helper for that case: https://github.com/status-im/status-go/blob/19771c422da460e870aa53701589be80cde1ff52/protocol/messenger_testing_utils.go#L53 but the PR with that is in deep draft. I'll use it here once I merge it.

Reply:
neat, we need to test more signals!